cmake/extensions.cmake

```diff
@@ -112,6 +126,11 @@ Please provide one of following: CONF_FILES")
   endif()
 
   zephyr_file(CONF_FILES ${FILE_CONF_FILES}/boards ${FILE_UNPARSED_ARGUMENTS})
+  if(ZEPHYR_FILE_DTS)
+    zephyr_file(CONF_FILES ${FILE_CONF_FILES}
+                ${ZEPHYR_FILE_UNPARSED_ARGUMENTS}
+                DTS ${ZEPHYR_FILE_DTS})
```

Comment:
This looks very wrong to me.

If user calls `ncs_file(CONF_FILES <path> ... DTS <list>)` then `zephyr_file()` will be invoked twice, and second time with `DTS ${ZEPHYR_FILE_DTS}` twice, because the `ZEPHYR_FILE_UNPARSED_ARGUMENTS` already contains `DTS ${ZEPHYR_FILE_DTS}` in its list because `DTS` is not present inside: `single_args`.

Reply:
Correct me if I'm wrong in my understanding of this script.

'ZEPHYR_FILE_UNPARSED_ARGUMENTS' will not hold 'DTS' due to `cmake_parse_arguments(ZEPHYR_FILE "" "KCONF;DTS;BUILD" "" ${ARGN})` (line 118).

The code, you are writing about, allows use of `DTS`, but it is optional. If `DTS` is provided by user, it will be parsed in line 118 and stored in `ZEPHYR_FILE_DTS` list. That list will be used by second call to `zephyr_file` to collect all devicetree overlays.  `${ZEPHYR_FILE_UNPARSED_ARGUMENTS}` is required here to provide any optional parameters like e.g. `BOARD`, `BOARD_REVISION`. 

In my humble opinion this is required part to enable reading devicetree overalys for child by the function.